pkg/apis/machine/types.go

```diff
@@ -492,10 +496,27 @@ const (
 
 	// RollingUpdateMachineDeploymentStrategyType means that old MCs will be replaced by new one using rolling update i.e gradually scale down the old MCs and scale up the new one.
 	RollingUpdateMachineDeploymentStrategyType MachineDeploymentStrategyType = "RollingUpdate"
+
+	// InPlaceUpdateMachineDeploymentStrategyType signifies that machines will be updated in place,
+	// gradually transitioning from the old MachineSet to the new MachineSet without requiring machine recreation.
+	InPlaceUpdateMachineDeploymentStrategyType MachineDeploymentStrategyType = "InPlaceUpdate"
 )
 
 // RollingUpdateMachineDeployment specifies the spec to control the desired behavior of rolling update.
 type RollingUpdateMachineDeployment struct {
+	UpdateConfiguration
+}
+
+// InPlaceUpdateMachineDeployment specifies the spec to control the desired behavior of inplace rolling update.
+type InPlaceUpdateMachineDeployment struct {
+	UpdateConfiguration
+
+	// OrchestrationType specifies the orchestration type for the inplace update.
+	OrchestrationType OrchestrationType
+}
+
+// UpdateConfiguration specifies the udpate configuration for the rolling strategy.
+type UpdateConfiguration struct {
```

Comment:
Since `UpdateConfiguration` is now part of both `Inplace` and `Rolling` strategies, can you make sure that the comments on the respective fields inside it are not `Rolling` specific?